apps/omg/lib/omg/state/measure.ex

```diff
@@ -27,9 +27,14 @@ defmodule OMG.State.Measure do
   def supported_events, do: @supported_events
 
   def handle_event([:process, OMG.State], _, %Core{} = state, _config) do
-    Enum.each(MeasurementCalculation.calculate(state), fn
-      {key, value} -> _ = Datadog.gauge(name(key), value)
-      {key, value, metadata} -> _ = Datadog.gauge(name(key), value, tags: [metadata])
-    end)
+    execute = fn ->
+      Enum.each(MeasurementCalculation.calculate(state), fn
+        {key, value} -> _ = Datadog.gauge(name(key), value)
+        {key, value, metadata} -> _ = Datadog.gauge(name(key), value, tags: [metadata])
+      end)
+    end
+
+    _ = Task.start(execute)
```

Comment:
three things to consider:
1/ leave a comment here telling our future selves that we're sending a beefy term over, we acknowledge it and ask for forgiveness. Also one word why we're doing it like that (to not block the owner of `state`)
2/ pattern match `{:ok, _} = `. Not sure about this, but intuitively I think I'd prefer this to crash if task cannot be started. I'm assuming such crash would have us notified but wouldn't bring everything else down
3/ `start_link` maybe? for same reasons as 2/ - reasoning: previously if `.calculate` would fail, this process would crash.

Reply:
> leave a comment here

+1

> I'm assuming such crash would have us notified but wouldn't bring everything else down

In this particular case it would bring down State process. Same goes for linking.

Reply:
> In this particular case it would bring down State process. Same goes for linking.

ah doh, of course, brainfart on my part. It still might or might not make sense to crash that State in case of failure of either 2/ or 3/ - this is what used to be true for non-Task approach.

Leaving up to you, what's better.